src/libstore/build/local-derivation-goal.cc

```diff
@@ -1778,6 +1778,10 @@ void LocalDerivationGoal::runChild()
             if (personality(PER_LINUX32) == -1)
                 throw SysError("cannot set i686-linux personality");
         }
+        if (drv->platform == "armv7l-linux" || drv->platform == "armv6l-linux") {
+          if (personality(PER_LINUX32) == -1)
+            throw SysError("cannot set 32bit linux personality");
+        }
```

Comment:
This can be combined with the i686-linux case above it (i.e. `if (... || drv->platform == "armv7l-linux" || drv->platform == "armv6l-linux") { ... }`). Otherwise LGTM.